docs/docs/50-user-guide/60-reference-docs/80-webhook-receivers/generic.md

```diff
@@ -4,8 +4,267 @@ sidebar_label: Generic
 
 # Generic Webhook Receiver
 
+The generic webhook receiver responds to any inbound POST request by determining
+whether it meets user-defined criteria, then executing user-defined actions on a
+user-defined set of resources when it does.
+
+:::note
+Currently, these actions are limited to "refreshing" `Warehouse` resources,
+which triggers their artifact discovery processes, so a typical use of this
+component is responding to "push" events from artifact repositories that
+lack dedicated webhook receiver implementations. Since this component
+effectively enables imperatively refreshing a `Warehouse` from any external
+process, other uses are possible and practical.
+:::
+
+## Configuring the Receiver
+
+A Generic webhook receiver must reference a Kubernetes `Secret` resource with
+a `secret` key in its data map.
+
 :::info
-__Coming soon!__
+Kargo incorporates the secret into the generation of a hard-to-guess URL for the
+receiver. This URL serves as a _de facto_
+[shared secret](https://en.wikipedia.org/wiki/Shared_secret) and authentication
+mechanism.
+:::
+
+:::note
+The following commands are suggested for generating and base64-encoding a
+complex secret:
+
+```shell
+secret=$(openssl rand -base64 48 | tr -d '=+/' | head -c 32)
+echo "Secret: $secret"
+echo "Encoded secret: $(echo -n $secret | base64)"
+```
+
+:::
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: wh-secret
+  namespace: kargo-demo
+  labels:
+    kargo.akuity.io/cred-type: generic
+data:
+  secret: <base64-encoded secret>
+---
+apiVersion: kargo.akuity.io/v1alpha1
+kind: ProjectConfig
+metadata:
+  name: kargo-demo
+  namespace: kargo-demo
+spec:
+  webhookReceivers:
+    - name: my-receiver
+      generic:
+        secretRef:
+          name: wh-secret
+```
+
+### whenExpression
```

Comment:
If I were coming into this with no prior knowledge, I think I'd benefit from a higher-level explanation of actions before diving into individual fields. Understanding that actions are defined by a type, an expression that determines if the action should be executed, and criteria that select resources the action should be applied to seems important.

I also think that before talking about `whenExpression`, `actionType` is necessary to talk about.

Reply:
In fact, I think because the amount of configuration possible here is much greater than for any other webhook receiver (which all basically require no configuration apart from a reference to a Secret), we might want to divide "Configuring the Receiver" into two main sections: Perhaps "Base Configuration" covering the same old boilerplate as all the webhook receivers required, followed by a "Defining Actions" section, which is where the missing high-level explanation of actions I mentioned in the previous comment would make some nice introductory text.